logstash-core/src/test/java/org/logstash/secret/store/backend/JavaKeyStoreTest.java

```diff
@@ -0,0 +1,275 @@
+package org.logstash.secret.store.backend;
+
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.logstash.secret.SecretIdentifier;
+import org.logstash.secret.store.SecretStoreException;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFileAttributeView;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static java.nio.file.attribute.PosixFilePermission.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Fail.fail;
+import static org.hamcrest.CoreMatchers.instanceOf;
+
+/**
+ * Unit tests for the {@link JavaKeyStore}
+ */
+public class JavaKeyStoreTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+    private JavaKeyStore keyStore;
+    private char[] keyStorePass;
+    private Path keyStorePath;
+
+    /**
+     * Simple example usage.
+     */
+    @Test
+    public void basicTest() {
+        String password = "pAssW3rd!";
+        //persist
+        keyStore.persistSecret(new SecretIdentifier("mysql.password"), password.getBytes(StandardCharsets.UTF_8));
+        //retrieve
+        byte[] secret = keyStore.retrieveSecret(new SecretIdentifier("mysql.password"));
+        assertThat(new String(secret, StandardCharsets.UTF_8)).isEqualTo(password);
+        //purge
+        keyStore.purgeSecret(new SecretIdentifier("mysql.password"));
+        secret = keyStore.retrieveSecret(new SecretIdentifier("mysql.password"));
+        assertThat(secret).isNull();
+    }
+
+    /**
+     * Tests that the magic marker that identifies this a logstash keystore is present.  This marker helps to ensure that we are only dealing with our keystore, we do not want
+     * to support arbitrary keystores.
+     *
+     * @throws Exception when ever it wants to.
+     */
+    @Test
+    public void isLogstashKeystore() throws Exception {
+        //newly created
+        byte[] marker = keyStore.retrieveSecret(new SecretIdentifier(JavaKeyStore.LOGSTASH_MARKER));
+        assertThat(new String(marker, StandardCharsets.UTF_8)).isEqualTo(JavaKeyStore.LOGSTASH_MARKER);
+
+        //exiting
+        JavaKeyStore existingKeyStore = new JavaKeyStore(Paths.get(this.getClass().getClassLoader().getResource("logstash.keystore").toURI()), "mypassword".toCharArray());
```

Comment:
:) already fixed in the next PR. (won't change in this PR since the constructor changes in the next PR and don't want to make a test code change just for this PR)